mbf_costmap_nav/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.0.2)
 project(mbf_costmap_nav)
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Q / F: MBF uses [C++03](https://en.wikipedia.org/wiki/C%2B%2B03) standard.

Do we need any C++11 or posterior feature?

I guess for master is just fine to use C++17. @spuetz , @RainerKuemmerle , thoughts on this?
And can we just set the standard on the metapackage?

Reply:
I am also using `std::optional` (c++17)

Reply:
c++17 is ok for me.